src/credentials/auth.ts

```diff
@@ -586,6 +587,16 @@ export class Auth implements AuthService, ConnectionManager {
         return this.#validationErrors.get(connection.id)
     }
 
+    public addTempCredentials(data: StaticProfile): CredentialsId {
+        const tempProvider = new TempCredentialsProvider(data)
+        this.iamProfileProvider.addProvider(tempProvider)
+        return tempProvider.getCredentialsId()
+    }
+
+    public removeTempCredentials(id: CredentialsId) {
+        this.iamProfileProvider.removeProvider(id)
+    }
```

Comment:
Exposing methods to temporarily change state in order to do other operations is a sign that these methods shouldn't be public. It would be better to consolidate them into a single method. Callers shouldn't need to worry about "temporary" state.

Reply:
Good point, I updated this commit and made a single public method in the Auth class that uses the above methods which are now private.